jitbuilder/runtime/JBCodeCacheMemorySegment.cpp

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2023
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+
+#include "env/CompilerEnv.hpp"
+#include "runtime/CodeCacheMemorySegment.hpp"
+
+
+void
+JitBuilder::CodeCacheMemorySegment::setSegmentBase(uint8_t *newBase)
```

Comment:
Is there a reason why this functionality was added to the JitBuilder class as opposed to the base OMR class?  I believe the base OMR class assumes these writes will be occurring in the code cache so it would make sense for this toggling of permissions to occur there.  The disable/enable calls themselves are macros that are only defined on Arm macOS, so there shouldn't be any performance issues with that.  Also, if a consuming implementation wishes to change this behaviour then the onus should be on that project to extend this class and define new behaviour.

Reply:
I agree: it doesn't seem like JitBuilder needs to extend the base class implementation (presumably the base class needs to call these `omrthread_jit_write_protect*` calls too?)

Reply:
I did not think that the OMR side assumed that the `CodeCacheMemorySegment` object would be allocated on the executable memory because the OMR’s `CodeCacheManager` allocates it with `getMemory()` API in `OMR::CodeCacheManager::setupMemorySegmentFromRepository.` Having said that, I agree that having the disable/enable calls on the OMR side would be appropriate considering the fact that those calls do not have performance issues as you pointed out. I will move them to the `OMR::CodeCacheMemorySegment` class.

Reply:
Updated the commit. Can you please review again? Thanks.

Reply:
@0xdaryl @mstoodle Is it OK to merge this PR?